src/main/java/net/imagej/ops/filter/bilateral/DefaultBilateral.java

```diff
@@ -0,0 +1,148 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imagej.ops.filter.bilateral;
+
+import net.imagej.ops.Ops;
+import net.imagej.ops.special.computer.AbstractUnaryComputerOp;
+import net.imglib2.Cursor;
+import net.imglib2.FinalInterval;
+import net.imglib2.Interval;
+import net.imglib2.RandomAccess;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.algorithm.neighborhood.Neighborhood;
+import net.imglib2.algorithm.neighborhood.RectangleNeighborhood;
+import net.imglib2.algorithm.neighborhood.RectangleNeighborhoodFactory;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.view.Views;
+
+import org.scijava.Priority;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+/**
+ * Performs a bilateral filter on an image. The sigmaS parameter refers to the
+ * spatial smoothing parameter; the greater the sigma, the smoother the image.
+ * The sigmaR parameter refers to the range smoothing parameter. The greater the
+ * sigma, the greater the effect of intensity differences. The radius parameter
+ * refers to the square that is considered when doing the filter on each
+ * individual picture.
+ *
+ * @author Gabe Selzer
+ * @param <I>
+ * @param <O>
+ */
+
+@Plugin(type = Ops.Filter.Bilateral.class, priority = Priority.NORMAL_PRIORITY)
+public class DefaultBilateral<I extends RealType<I>, O extends RealType<O>>
+	extends
+	AbstractUnaryComputerOp<RandomAccessibleInterval<I>, RandomAccessibleInterval<O>>
+	implements Ops.Filter.Bilateral
+{
+
+	public final static int MIN_DIMS = 2;
+
+	public final static int MAX_DIMS = 2;
+
+	@Parameter
+	private double sigmaR;
+
+	@Parameter
+	private double sigmaS;
+
+	@Parameter
+	private int radius;
+
+	private static double gauss(final double x, final double sigma) {
+		final double mu = 0.0;
+		return (1 / (sigma * Math.sqrt(2 * Math.PI))) * Math.exp((-0.5 * (x - mu) *
+			(x - mu)) / (sigma * sigma));
+	}
+
+	@Override
+	public void compute(final RandomAccessibleInterval<I> input,
+		final RandomAccessibleInterval<O> output)
+	{
+		if (input.numDimensions() != 2) {
+			throw new IllegalArgumentException("Input must be two dimensional");
+		}
+
+		final long[] size = new long[input.numDimensions()];
+		input.dimensions(size);
+
+		final RandomAccess<O> cr = output.randomAccess();
```

Comment:
Variables could be a little more descriptive, especially since there are so many similar ones in the implementation.